content/blog/2022-12-06-keda-2.9.0-release.md

```diff
@@ -0,0 +1,44 @@
++++
+title = "Announcing KEDA v2.9 🎉"
+date = 2022-12-08
+author = "Jeff Hollan (KEDA), Tom Kerkhove (KEDA)"
+aliases = [
+"/blog/releases/2.9.0"
+]
++++
+
+We recently completed our most recent release: 2.9.0 🎉!
+
+Here are some highlights:
+
+- Newly published Deprecations and Breaking Change policy ([docs](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)) - this includes enforcing our Kubernetes compatability policy (meaning users will need to use Kubernetes >=1.23 with version 2.9.0)
+- New Loki Scaler ([docs](https://keda.sh/docs/scalers/loki/))
+- All metrics exposed to prometheus are now consolidated in KEDA Operator, including some new metrics
+- Azure Key Vault authentication provider now supports pod identities for authentication
+- A ton of new features and fixes for some of our 50+ scalers
+
+Potential breaking changes and deprecations include:
+- As per our [support policy](https://github.com/kedacore/governance/blob/main/SUPPORT.md), Kubernetes v1.23 or above is required and support for Kubernetes v1.22 or below was removed ([docs](https://keda.sh/docs/2.9/operate/cluster/#kubernetes-compatibility))
+- Upgrading the API version of managed HPA objects from `autoscaling/v2beta2` to `autoscaling/v2`
+- Prometheus metrics on KEDA Metric Server are deprecated in favor of Prometheus metrics on KEDA Operator ([#3972](https://github.com/kedacore/keda/issues/3972) | [Discussion](https://github.com/kedacore/keda/discussions/3973))
+
+You can keep an eye and join the discussion for future breaking changes and deprecations [in our GitHub issues](https://github.com/kedacore/keda/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Abreaking-change).
+
```

Comment:
```suggestion
## Change in the architecture and metrics caching

In this release we focused on performance improvements. Let's talk about this changes: KEDA has two main components - the Operator and the Metrics Server. The Operator monitors ScaledObjects and ScaledJobs resources, and checks activity on the external service. The Metrics Server handles requests and queries about metric values used for scaling from Kubernetes' Horizontal Pod Autoscaler (HPA) by checking the external service.

In previous versions of KEDA, each component needed to handle its own connection to the external service. In version 2.9, we have changed this and moved the connection logic to the Operator component. We are now using gRPC for the connection between the Metrics Server and Operator. This has several benefits, including reduced load on the external service, as there is only one connection from the Operator.

We have also introduced an experimental feature called metrics value caching. This caches the values in the Operator and can be reused in future. It can even further reduce the load on the external service. By default, Kubernetes' HPA asks for a metric value every few seconds (as defined b `--horizontal-pod-autoscaler-sync-period` flag set by Kuberentes admins, which is usually set to 15 seconds). This can create unnecessary load on the external service. Caching metric values per trigger in a ScaledObject can help to overcome this problem. This is an optional feature, and can be enabled by setting the `useCachedMetrics` flag, then metric values are cached during polling interval as defined by `pollingInterval` flag and request comming from HPA to KEDA Metrics Server are served from this cache.

We are excited about the potential uses for this feature in the future, including using AI and machine learning models to affect metric values. Stay tuned for more updates and improvements in future releases of KEDA.
```